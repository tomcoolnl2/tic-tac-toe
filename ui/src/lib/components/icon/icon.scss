.icon {
	padding: 4px;
	cursor: pointer;
}
