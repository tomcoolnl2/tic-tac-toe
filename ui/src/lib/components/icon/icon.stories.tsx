import type { Meta } from '@storybook/react';
import { Icon } from './icon';

const Story: Meta<typeof Icon> = {
	component: Icon,
	title: 'Components/Icon',
};
export default Story;

export const Default = {
	args: {
		name: 'icon-repeat',
	},
};
