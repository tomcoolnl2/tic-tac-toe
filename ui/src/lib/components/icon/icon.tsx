import React from 'react';
import './icon.scss';

export interface Props {
	name: string;
	testId?: string;
	handleOnClick?: () => void;
}

export const Icon: React.FC<Props> = React.memo(({ name, testId, handleOnClick }) => {
	return <i className={`icon ${name}`} onClick={() => handleOnClick?.()} data-testid={testId} />;
});
