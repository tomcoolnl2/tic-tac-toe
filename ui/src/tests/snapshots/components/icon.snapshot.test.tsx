import { render } from '@testing-library/react';
import { Icon } from '../../../lib/components/icon/icon';

describe('Icon component', () => {
	it('matches snapshot', () => {
		const { container } = render(<Icon name="icon-pause" handleOnClick={() => ({})} />);
		expect(container.firstChild).toMatchSnapshot();
	});
});
