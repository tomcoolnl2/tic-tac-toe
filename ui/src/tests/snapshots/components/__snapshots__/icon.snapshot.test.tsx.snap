// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component matches snapshot 1`] = `
<i
  class="icon icon-pause"
/>
`;
